src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Utility methods and constants for manipulating Configuration paths
+    /// </summary>
+    public static class ConfigurationPath
+    {
+        public static readonly string KeyDelimiter = ":";
+
+        public static string Combine(params string[] pathSegements)
+        {
+            if (pathSegements == null)
+            {
+                throw new ArgumentNullException(nameof(pathSegements));
+            }
+            return string.Join(KeyDelimiter, pathSegements);
+        }
+
+        public static string Combine(IEnumerable<string> pathSegements)
+        {
+            if (pathSegements == null)
+            {
+                throw new ArgumentNullException(nameof(pathSegements));
+            }
+            return string.Join(KeyDelimiter, pathSegements);
+        }
+
+        public static string GetLastSegment(string path)
```

Comment:
There's literally two things we do with configuration paths right now,  build one...  and extract the key.  We are already being a bit minimalist in that the operation that's missing is the getBasePath equivalent which is the path minus the last segment...

But, I personally hate writing parsing code like this, inflicting that on everyone seems unnecessary painful when we can do it once (hopefully correctly :)